### PR TITLE
CHANGE(BMP-612): add logging functionality, fix routes exposure bug and refactor for new pipeline

### PR DESCRIPTION
**Pull request by FilippoRezzonico**
I have modified the repo directory names and refactored some files to be compatible with the new pipeline for .NET templates and examples. 
The necessary files to enable logging from the template have been added. 
This template now shows status routes in the api portal because the swagger documentation passed to the library has been fixed by passing the `Version` property. 

#### Checklist

[BMP-612](https://makeitapp.atlassian.net/browse/BMP-612)

- [ ] the documentation is updated or integrated accordingly with your changes
- [X] you are not committing extraneous files or sensible data
- [X] added the link to the Jira task
- [X] commit message and branch name conventions

#### Notify to

@fredmaggiowski

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ namespace DotNetCore_Template.Controllers
 }
 ```
 
-Visiting the defined route: **/hello** through a **GET** request, you will receive the `Hello World!` message as response. 
+Visiting the defined route: **/hello** through a **GET** request, you will receive the `Hello World!` message as response.
```
Comment (fredmaggiowski) [inline README.md:99]: This line is useless, there should be no `/hello` handler

File: src/App/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1 @@
+[assembly: log4net.Config.XmlConfigurator(ConfigFile = "log4net.config")]
```
Comment (fredmaggiowski) [inline src/App/Properties/AssemblyInfo.cs]: style: newline

```suggestion
[assembly: log4net.Config.XmlConfigurator(ConfigFile = "log4net.config")]

```

File: src/App/log4net.config [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    <root>
+      <level value="ALL" />
+      <appender-ref ref="console" />
+    </root>
+    <appender name="console" type="log4net.Appender.ConsoleAppender">
+      <layout type="log4net.Layout.PatternLayout">
+        <conversionPattern value="%message%newline" />
+      </layout>
+    </appender>
+  </log4net>
```
Comment (fredmaggiowski) [inline src/App/log4net.config]: style: newline 

```suggestion
  </log4net>

```

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (FilippoRezzonico) [reply README.md:99]: Actually this part of the readme explains how to add an hello handler to the template! Therefore, I think that this line is correct!

File: src/App/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (FilippoRezzonico) [reply src/App/Properties/AssemblyInfo.cs]: Added missing newline

File: src/App/log4net.config (hunk 3 above)
Comment (FilippoRezzonico) [reply src/App/log4net.config]: Added missing newline

File: README.md (hunk 1 above)
Comment (fredmaggiowski) [reply README.md:99]: ok fine

**Review by fredmaggiowski — APPROVED** [review]

**Pull request merged**